tools/lsp/properties.rs

```diff
@@ -491,30 +491,32 @@ fn get_properties(
     result
 }
 
-fn find_block(token: Option<i_slint_compiler::parser::SyntaxToken>) -> Option<(u32, u32)> {
+fn find_block(
+    token: Option<i_slint_compiler::parser::SyntaxToken>,
+) -> Option<(i_slint_compiler::parser::SyntaxToken, i_slint_compiler::parser::SyntaxToken)> {
     let mut current_token = token;
     let mut brace_level = 0_u32;
-    let mut start_offset = u32::MAX;
-    let mut end_offset = u32::MAX;
+    let mut start_token = None;
+    let mut end_token = None;
 
     while let Some(t) = current_token {
         if t.kind() == SyntaxKind::LBrace {
             brace_level += 1;
             if brace_level == 1 {
-                start_offset = t.text_range().end().into();
+                start_token = Some(t.clone());
             }
         } else if t.kind() == SyntaxKind::RBrace {
             brace_level -= 1;
             if brace_level == 0 {
-                end_offset = Into::<u32>::into(t.text_range().start());
+                end_token = Some(t.clone());
                 break;
             }
         }
         current_token = t.next_token();
     }
 
-    if start_offset != u32::MAX && start_offset <= end_offset {
-        return Some((start_offset, end_offset));
+    if start_token.is_some() && end_token.is_some() {
```

Comment:
That looks like Option::zip
otherwise, please use `if let` instead of is_some then expect.